webapp/src/components/news-item/NewsItem.test.js

```diff
@@ -0,0 +1,139 @@
+import { describe, expect, it } from '@jest/globals';
+import { mount } from '@vue/test-utils';
+import NewsItem from '@/components/news-item/NewsItem';
+
+require('regenerator-runtime/runtime');
+
+
+const TITLE = 'Test-Title';
+const VOTES = 0;
+
+describe('NewsItem', () => {
+
+  it('should set news from parent as props', () => {
+    const title = 'Test-Title';
+    const votes = 0;
+    const wrapper = mount(NewsItem, {
+      propsData: {
+        newsItem: {
+          title: title,
+          votes: votes,
+        },
+      },
+    });
+
+    expect(wrapper.props().newsItem.title).toBe(title);
+    expect(wrapper.props().newsItem.votes).toBe(votes);
+
+  });
+
+  it('should show a h1 with news-title/news-votes and three buttons for upVote, downVote and remove', () => {
+    const title = 'Test-Title';
+    const votes = 0;
+    const wrapper = mount(NewsItem, {
+      propsData: {
+        newsItem: {
+          title: title,
+          votes: votes,
+        },
+      },
+    });
+
+    const expectedText = title + ' ' + votes;
+
+    const actualH1 = wrapper.find('h1');
+    const actualUpvoteButton = wrapper.find('button.upvote');
+    const actualDownvoteButton = wrapper.find('button.downvote');
+    const actualRemoveButton = wrapper.find('button.remove');
+
+    expect(actualH1).toBeDefined();
+    expect(actualH1.text()).toEqual(expectedText);
+
+    expect(actualUpvoteButton).toBeDefined();
+    expect(actualDownvoteButton).toBeDefined();
+    expect(actualRemoveButton).toBeDefined();
+
+  });
+
+  it('should emit upvoting-updateNews event once upvote-button is clicked', async () => {
+    const wrapper = mount(NewsItem, {
+      propsData: {
+        newsItem: {
+          title: TITLE,
+          votes: VOTES,
+        },
+      },
+    });
+
+    const button = wrapper.find('button.upvote');
+    button.trigger('click');
+
+    await wrapper.vm.$nextTick();
+
+    const expectedCallCount = 1;
+    const upvoteEvent = wrapper.emitted().updateNews;
+
+    expect(upvoteEvent).toBeTruthy();
+    expect(upvoteEvent.length).toBe(expectedCallCount);
+
+    const actualUpdatedNews = upvoteEvent[0][0];
+    expect(actualUpdatedNews.title).toEqual(TITLE);
+    expect(actualUpdatedNews.votes).toEqual(VOTES + 1);
+  });
+
+  it('should emit downvoting-updateNews event once downvote-button is clicked', async () => {
+    const wrapper = mount(NewsItem, {
+      propsData: {
+        newsItem: {
+          title: TITLE,
+          votes: VOTES,
+        },
+      },
+    });
+
+    const button = wrapper.find('button.downvote');
+    button.trigger('click');
+
+    await wrapper.vm.$nextTick();
+
+    const expectedCallCount = 1;
```

Comment:
```suggestion
```
remove and use a hard-coded value on line 103

Reply:
I find it more readable with a variable.